app/Http/Controllers/VoteController.php

```diff
@@ -188,25 +190,67 @@ public function getUserVote($userId, $voteId)
      * @param $voteId
      * @return \Illuminate\Http\JsonResponse
      */
-    public function getUserVoteResult($voteId)
+    public function getUserVoteResult(Vote $vote)
     {
-        $vote = Vote::findOrFail($voteId);
-        $voteResults = $vote->voteResults()->get();
-
-        if (!$voteResults) {
-            throw (new ModelNotFoundException)->setModel(VoteResult::class);
+        $user = Auth::user();
+        $voteItems = $vote->voteItems()->get();
+        if (!$voteItems) {
+            throw (new ModelNotFoundException)->setModel(VoteItem::class);
+        }
+        $meta['checked'] = [];
+        $userVoteResults = $vote->voteResults()->where('user_id', $user->id)->get();
+        foreach ($userVoteResults as $res) {
+            $temp = $voteItems->where('id', $res->vote_item_id);
+            foreach ($temp as $item) {
+                $item->checked = 1;
+            }
         }
+        $meta['vote'] = $vote;
+        return $this->setStatusCode(200)->respond($voteItems, $meta);
 
-        return $this->setStatusCode(200)->respond($voteResults, ['vote' => $vote]);
     }
 
     /**
      * Display the specific vote all results
      * @return \Illuminate\Http\JsonResponse
      */
-    public function createUserVoteResult(VoteResultRequest $request)
-    {
-        $voteresult = VoteResult::create($request->all());
-        return $this->setStatusCode(201)->respond($voteresult);
+    public function createUserVoteResult($id,
+        VoteResultRequest $request
+    ) {
+        $model = null;
+        $vote = Vote::findOrFail($request->vote_id);
+        $user = Auth::user();
+        $voteItem = VoteItem::findOrFail($request->vote_item_id);
+        $response['checked'] = true;
+        if ($vote->is_single) {
+            $model = $vote->voteResults()->where('user_id', $user->id)->get();
```

Comment:
Probably you chose not so suitable variable name. Because I guess you will get an array of models here:
$model = $vote->voteResults()->where('user_id', $user->id)->get();